executor/delete_test.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2020 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package executor_test
+
+import (
+	"sync"
+	"time"
+
+	. "github.com/pingcap/check"
+	"github.com/pingcap/tidb/util/testkit"
+)
+
+func (s *testSuite8) TestDeleteLockKey(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec("use test")
+
+	tk.MustExec(`drop table if exists t1, t2, t3;`)
+
+	cases := []struct {
+		ddl     string
+		pre     string
+		tk1Stmt string
+		tk2Stmt string
+	}{
+		{
+			"create table t1(k int, kk int, val int, primary key(k, kk), unique key(val))",
```

Comment:
We could add some cases to cover delete lock behaviours for other executors like `batchPointGet`, `tableScan`.